core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -421,10 +418,9 @@ class ExecutorSuite extends SparkFunSuite
 
   private def createResultTaskDescription(
       serializer: SerializerInstance,
-      resultFunc: (TaskContext, Iterator[Int]) => Int,
+      taskBinary: Broadcast[Array[Byte]],
       rdd: RDD[Int],
       stageId: Int): TaskDescription = {
-    val taskBinary = sc.broadcast(serializer.serialize((rdd, resultFunc)).array())
```

Comment:
I'm curious why you decided to move this out of the helper?

Reply:
I don't actually understand why, but when the call to `sc.broadcast` happens inside the helper, there were some problems with the broadcast. If the entire `ExecutorSuite` were run, all the tests would pass. However, if I ran each test in it individually, then the tests that used this helper would fail, due to said problem.